ui: collect statement stats from all nodes

couchand opened the pull request:
The initial statement stats API endpoint only exposed the local node's statements.  This extends it to query all the other nodes as well, properly aggregates across nodes, and provides the per-node details on the statement details page.

<img width="659" alt="screen shot 2018-06-28 at 5 19 00 pm" src="https://user-images.githubusercontent.com/793969/42061214-65fe50bc-7af7-11e8-81f1-839256985e03.png">

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/26605)
<!-- Reviewable:end -->

couchand commented:
I think this is ready to go @vilterp PTAL

couchand commented:
Oh I still need to aggregate the results...

couchand commented:
Probably still want to add a proper test for this, but it's working!

vilterp commented on pkg/server/serverpb/status.proto:
```diff
@@ -608,6 +608,7 @@ message StoresResponse {
 }
 
 message QueriesRequest {
+  string node_id = 3 [ (gogoproto.customname) = "NodeID" ];
```
nit: no spaces inside `[ ]`s

vilterp reviewed (commented):
Looks good; add a test and :shipit:

couchand committed:
server, ui: request statement statistics from all nodes

This relays the Queries RPC to all the other nodes in the cluster to
enable the Statements page to show the statistics gathered on all
nodes rather than just the gateway.

Fixes: #26240
Release note (admin ui change): The Statements page now shows statements
that executed on all nodes in the cluster, not just the gateway node.

couchand committed:
server: move queries rpc from admin to status server

This will enable us to use the iterateNodes helper to fan out the request
to all the nodes in the cluster.

Release note: None

couchand committed:
server, ui: rename Queries RPC to Statements

To keep the terminology consistent.

Release note: None

couchand committed:
ui: pass node id through all statement stats aggregation

Release note: None

couchand committed:
ui: extract statements table component

Release note: None

couchand committed:
ui: add node breakdown table to statment details page

Release note (admin ui change): This adds to the Statement Details page a table
showing the statistics broken down by which node was the gateway node.

couchand commented:
This has now been rebased on top of both #26949 and #26821, only the last six commits should be reviewed here.  This should be ready for another look @vilterp, maybe @knz since I made use of your refactor, and anyone else who would like to weigh in here.

vilterp commented:
Noice! 
![image](https://user-images.githubusercontent.com/7341/42062542-d5ab99b6-7afb-11e8-913d-b98a9e54bd26.png)

Layout suggestion to save a bit of width: show just the node ids ("n1", etc) and put the host and port in a tooltip? Will be esp. helpful when hostnames are longer, which they often are (`cockroach.my-kubernetes-cluster.my-co.internal`...)

Also, nit: a couple of pixels under the h3s (between them and the tables) would be nice.

knz commented:
I would recommend holding on merging this until #26821 merges, since Ben made a recommendation about a tweak in the API and I still need to process that recommendation.

vilterp commented on pkg/server/status_test.go:
```diff
@@ -849,3 +852,72 @@ func TestRemoteDebugModeSetting(t *testing.T) {
 	}
 }
+
+func TestStatusAPIQueries(t *testing.T) {
```
I guess this should be called `TestStatusAPIStatements`? (Thx for doing the rename btw)

vilterp reviewed (approved):
LGTM with a minor nit. Tested locally; code looks good.

couchand commented:
Rebased on master, last few comments addressed, and ready to go when I see a green build I think.

couchand commented:
ok, ok let's just go

bors r+

couchand commented:
bors r+
